### PR TITLE
refactor(fire): apply the ADR-0007 day policy through one shared helper

### LINKED ISSUE #901: TIDY_UP-08: fire.py package split and one gap-policy implementation (ADR-0007)
## Files
- `src/climate_indices/fire.py` — KBDI (:191-1213), CFFWIS moisture codes (:1213-2132), Fosberg (:2132-2214), HDW (:2214-2721); `_apply_gap_policy` (:285-320); `_run_cffwis_recurrence` (:1451-1540)
- `docs/adr/0006-fire-recursive-state-and-execution.md`, `docs/adr/0007-fire-missing-data-policy.md`

## Problem
- KBDI's daily loop (`fire.py:752-772`) inlines a verbatim copy of `_apply_gap_policy` (`fire.py:299-320`), the ADR-0007 implementation. The CFFWIS codes go through `_run_cffwis_recurrence` → `_apply_gap_policy` (`fire.py:1489-1492`). Any ADR-0007 policy fix must now be made twice; the copies can drift silently (no test compares them).
- The 2721-line module holds four unrelated index families plus shared recurrence helpers. Navigation and review cost is high; section comments are the only structure.

## Solution
1. KBDI calls `_apply_gap_policy` per day instead of inlining it; its loop keeps its own wet-spell state (ADR-0006 keeps per-index state dataclasses).
2. Split into a `fire/` package: `kbdi.py`, `cffwis.py`, `fosberg.py`, `hdw.py`, `_recurrence.py`, with `fire/__init__.py` re-exporting the current public names so `from climate_indices import fire; fire.kbdi(...)` is unchanged.

Step 1 is the low-risk, behavior-preserving part; do it first.

## Benefits
Locality: ADR-0007 has one implementation. Leverage: package seams let CFFWIS/KBDI/Fosberg/HDW evolve and be tested independently instead of editing a 2.7k-line module.

## Recommendation strength
**Worth exploring** (the gap-policy dedupe inside is **Strong**)

## Related
#803 (closed; CFFWIS moisture codes), #850 (closed; removed dead scaffold), #885 (TIDYUP-1), #812 (fire benchmarks)

## Acceptance
- KBDI outputs bit-identical on existing fixtures; a test asserts the shared helper is the only gap-policy implementation.
- `uv run ruff check src/ tests/`, `uv run ruff format --check src/ tests/`, `uv run mypy src/`, `uv run pytest` pass.

### PR DESCRIPTION
**Pull request by monocongo**
Closes #901.

## Summary

Completes the remaining item in #901: KBDI's daily loop inlined a verbatim copy of `_apply_gap_policy` from `fire/_common.py`, the ADR-0007 implementation that the CFFWIS codes already went through. Any policy fix had to be made twice, with no test comparing the copies. KBDI now calls the shared helper (no `in_season` mask); behavior is unchanged.

The package-split step of #901 already landed in #895 (ADR-0005's line-count promotion), with a deliberate layout deviation recorded in `docs/design/fire-subsystem.md`: underscored implementation modules so `fire.kbdi`/`fire.cffwis` stay functions, and no separate `_recurrence.py`. This PR is the low-risk behavior-preserving dedupe only.

## Tests

`test_missing_day_policy_routes_through_the_shared_helper` spies on `_apply_gap_policy` from both the KBDI path (`tests/test_fire_kbdi.py`) and the CFFWIS paths (`tests/test_fire_cffwis_moisture.py`, parametrized over ffmc/dmc/dc). Either path re-inlining the policy fails the spy.

## Verification

- `uv run ruff check src/ tests/` — clean
- `uv run ruff format --check src/ tests/` — clean
- `uv run mypy src/ tests/test_type_checking.py` — clean
- `uv run pytest` — 2124 passed, 423 deselected
- `uv run pytest -m validation` — 378 passed, 3 skipped

## Related

- #901 — step 2 (package split) landed in #895; this is the step 1 gap-policy dedupe
- #799, #803 — the two implementations this collapses
- #885 — TIDY_UP work

### CODE REVIEW

**Comment by monocongo** [comment]
### 🛠️ Multi-Agent Review Summary
- **Review target:** https://github.com/monocongo/climate_indices/pull/984
- **Lanes run:** 3 read-only lanes — correctness/edge cases (deepseek-v4-pro), tests (deepseek-v4-pro), dataflow (deepseek-v4-pro). An initial dataflow dispatch on openai-codex/gpt-5.5 failed with a provider usage limit and was rerun successfully on deepseek-v4-pro.
- **Validation:** `uv run ruff check src/ tests/` clean; `uv run ruff format --check src/ tests/` clean; `uv run mypy src/ tests/test_type_checking.py` clean; `uv run pytest` 2125 passed, 423 deselected; `uv run pytest -m validation` 378 passed, 3 skipped.
- **Fix commits:** [cd4ada16 — test(fire): pin per-day routing and forwarded policy in the gap-policy spies](https://github.com/monocongo/climate_indices/commit/cd4ada16234aaf84aee0396fde901c0f3dd0c8b0)

### Findings and fix status

#### [P3] The gap-policy spies could not detect partial re-inline or wrong argument forwarding — Fixed
- **Location:** `tests/test_fire_kbdi.py` (567–585), `tests/test_fire_cffwis_moisture.py` (700–720)
- **Origin:** correctness, tests, dataflow lanes (one underlying finding)
- **Problem:** The recording helper collected only positional args and asserted `calls` was non-empty. A partial re-inline (helper called only on some days) or a wrong `nan_policy`/`max_gap_days` forward would still pass, and the CFFWIS case never exercised a non-default policy.
- **Applied fix:** Both tests now assert one helper call per simulated day; the KBDI case records keyword arguments, is parametrized over `propagate` and `bridge`, and asserts the forwarded policy values. Commit `cd4ada16`. Full gate rerun after the fix (results above).

#### [P4] Static-invalid cells are not separately pinned by the spies — Accepted, no change
- **Location:** `tests/test_fire_kbdi.py` (578), `tests/test_fire_cffwis_moisture.py` (715)
- **Origin:** tests lane
- **Problem:** Neither spy fixture includes a NaN static climatology, so that branch's routing is not asserted.
- **Applied fix:** None. The spies exist to pin that ADR-0007 has one implementation; the static-invalid semantics are already pinned behaviorally (`test_nan_mean_annual_precipitation_is_not_treated_as_a_missing_weather_day` in KBDI). The lane self-assessed the fix as low impact, so it is not pursued.

### Remaining blockers
None. The refactor is a semantic-preserving move (correctness and dataflow lanes found no parity or state defects), and all project checks pass.

File: tests/test_fire_kbdi.py [hunk 1]
```diff
@@ -564,6 +564,23 @@ def _gapped_series() -> tuple[np.ndarray, np.ndarray]:
...
+    def recording_helper(*args: object, **kwargs: object) -> object:
+        calls.append(args)
+        return shared(*args, **kwargs)
+
+    monkeypatch.setattr(fire._kbdi, "_apply_gap_policy", recording_helper)
+
+    precipitation, temperature = _gapped_series()
+    fire.kbdi(precipitation, temperature, 1000.0)
+
+    assert calls
```
[the remark this answers is not in the document]
Comment (monocongo) [reply tests/test_fire_kbdi.py]: Fixed in d954c449: both routing tests now capture the positional args passed to the recording spy (not just kwargs) and assert `day_weather_valid` is `False` on the fixture's poisoned day(s) and `True` on every other day, for both `fire._kbdi` and `fire._cffwis`. That closes the gap the finding identified — a variant that kept helper calls for valid days but branched locally on invalid ones would now fail these tests.
